sdks/java/core/src/test/java/org/apache/beam/sdk/testing/CoderPropertiesTest.java

```diff
@@ -213,4 +215,39 @@ public void testClosingCoderFailsWhenEncoding() throws Exception {
     expectedException.expectMessage("Caller does not own the underlying");
     CoderProperties.encode(new ClosingCoder(), Context.NESTED, "test-value");
   }
+
+  /** Coder that consumes more bytes while decoding than required. */
+  public static class BadCoderThatConsumesMoreBytes extends NonDeterministicCoder {
+
+    @Override
+    public String decode(InputStream inStream, Context context) throws IOException {
+      String value = super.decode(inStream, context);
+      byte[] extraBytes = new byte[Ints.BYTES];
```

Comment:
No point in having this array, can call inStream.read() which is guaranteed to return a byte or -1 for EOF